pynest/examples/hpc_benchmark.py

```diff
@@ -0,0 +1,426 @@
+   which the excitatory-excitatory neurons exhibit STDP with
+   multiplicative depression and power-law potentiation. A mutual
+   equilibrium is obtained between the activity dynamics (low rate in
+   asynchronous irregular regime) and the synaptic weight distribution
+   (unimodal). The number of incoming connections per neuron is fixed
+   and independent of network size (indegree=11250).
+
+   This is the standard network investigated in:
+   Morrison et al (2007). Spike-timing-dependent plasticity in balanced random
+     networks. Neural Comput 19(6):1437-67
+   Helias et al (2012). Supercomputers ready for use as discovery machines for
+     neuroscience. Front. Neuroinform. 6:26
+   Kunkel et al (2014). Spiking network simulation code for petascale
+     computers. Front. Neuroinform. 8:78
+
+   A note on scaling
+   -----------------
+
+   This benchmark was originally developed for very large-scale simulations on
+   supercomputers with more than 1 million neurons in the network and
+   11.250 incoming synapses per neuron. For such large networks, input to a
+   single neuron will be little correlated and network activity will remain
+   stable for long periods of time.
+
+   The original network size corresponds to a scale parameter of 100 or more.
+   In order to make it possible to test this benchmark script on desktop
+   computers, the scale parameter is set to 1 below, while the number of
+   11.250 incoming synapses per neuron is retained. In this limit, correlations
+   in input to neurons are large and will lead to increasing synaptic weights.
+   Over time, network dynamics will therefore become unstable and all neurons
+   in the network will fire in synchrony, leading to extremely slow simulation
+   speeds.
+
+   Therefore, the presimulation time is reduced to 50 ms below and the
+   simulation time to 250 ms, while we usually use 100 ms presimulation and
+   1000 ms simulation time.
+
+   For meaningful use of this benchmark, you should use a scale > 10 and check
+   that the firing rate reported at the end of the benchmark is below 10 spikes
+   per second.
+'''
+
+import numpy as np
+import os
+# from scipy.special import lambertw
+import sys
+import time
+
+import nest
+import nest.raster_plot
+
+M_INFO = 10
+M_ERROR = 30
+
+
+'''
+ Parameter section
+
+ define all relevant parameters: changes should be made here
+'''
+
+params = {
+    'nvp': 1,               # total number of virtual processes
+    'scale': 1.,           # scaling factor of the network size,
+                            # total network size = scale*11250 neurons
+    'simtime': 250.,        # total simulation time in ms
+    'presimtime': 50.,      # simulation time until reaching equilibrium
+    'dt': 0.1,              # simulation step
+    'record_spikes': True,  # switch to record spikes of excitatory neurons to file
+    'path_name': './',      # path where all files will have to be written
```

Comment:
I think the default path name (CWD) should be empty, since the expression for "this directory" may differ from `.` on non-Unix systems. The slash should definitely not be included.